frontend/docsettings.lua

```diff
@@ -154,6 +187,20 @@ function DocSettings:getFileFromHistory(hist_name)
     end
 end
 
+--- Returns the directory and full filepath of a hash-ID-based sidecar metadata store
+-- @string doc_path path to the document (e.g., `/foo/bar.pdf`)
+function DocSettings:getSidecarHashDirAndFilepath(doc_path)
+    -- Getting PDF ID from trailer via mupdf has not been implemented - everything uses partial MD5
+    local path = self:getSidecarDir(doc_path, "hash") -- could this be cached somehow?
+    local filetype = doc_path:match(".+%.(%w+)$")
+    if not filetype or filetype == "" then
+        return "",""
+    end
+    local hash_file = "metadata."..filetype..".lua"
+    local hash_filepath = path..'/'..hash_file
+    return path, hash_filepath
```

Comment:
Ah yes the "could this be cached somehow" discussion was forgotten. This would especially help if navigating through large directories more than once (like jumping in an out of large folders/subfolders) to speed up subsequent navigation. Latest commit will have an impl.